File: CursoPOOUber/PHP/uberVan.php
<?php
require_once('car.php');
class UberVan extends Car {
   public $typeCarAccepted;
    public $seatsMaterial;

    // public function __construct($license, $driver){
    //     parent::__construct($license,$driver);
    // }

    public function __construct($license, $driver, $typeCarAccepted, $seatsMaterial){
        parent::__construct($license,$driver);
        $this->typeCarAccepted = $typeCarAccepted;
        $this->seatsMaterial = $seatsMaterial;
    }


    public function setPassenger ($passenger){
        if($passenger == 6){
            $this->passenger =$passenger;
            $this->printDataCar();
        }else{
            echo "<br>UberVan --Necesitas asignar 6 pasajeros";
        }
    }

    public function printDataCar() {
        echo "<br> UberVan --Licencia: $this->license Driver: ".$this->driver->name ;
    }


}
?>

File: CursoPOOUber/Java/Java/UberVan.java
package Java;
import java.util.ArrayList;
import java.util.Map;


class UberVan extends Car{

Map<String, Map<String,Integer>> typeCarAccepted;
ArrayList<String> seatsMaterial;
Integer passenger;

public UberVan(String license, Account driver){
    super(license, driver);    

}

public UberVan(String license, Account driver,Map<String, Map<String,Integer>> typeCarAccepted,ArrayList<String> seatsMaterial){
    super(license, driver);
    this.typeCarAccepted = typeCarAccepted;
    this.seatsMaterial  = seatsMaterial;
    

}

@Override
public void setPassenger(Integer passengerUV) {
    if(passengerUV == 6){
        this.passenger=passengerUV;
        this.printDataCar();
    }else{
        System.out.println("Necesitas asignar 6 pasajeros");
    }

}

@Override
void printDataCar() {
    super.printDataCar();
    System.out.println(" Licencia : " + license + " Driver : " + driver.name);
    
}


}
